pkg/operator/controller/enablednsnameresolver/controller.go

```diff
@@ -95,21 +95,23 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 			// Start the dependent cache.
 			go func() {
 				if err := (*cache).Start(ctx); err != nil {
-					logrus.Error(err, "cannot start cache")
+					logrus.Errorf("failed to start cache, request.Name: %s, err: %v", request.Name, err)
 				}
 			}()
 
 			// Wait for the dependent cache to sync.
 			wg.Add(1)
 			go func() {
+				defer wg.Done()
 				if started := (*cache).WaitForCacheSync(ctx); !started {
 					logrus.Error("failed to sync cache before starting controllers")
 				}
-				wg.Done()
 			}()
 		}
 		// Wait for all the dependent caches to sync.
 		wg.Wait()
+
+		logrus.Info("dependent caches synced")
```

Comment:
This is misleading, because there could have been one that errored. Better not log success here.

```suggestion

```